src/main/java/frc/robot/hardware/mechanisms/MechanismSimulation.java

```diff
@@ -0,0 +1,30 @@
+package frc.robot.hardware.mechanisms;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+
+public abstract class MechanismSimulation {
+
+	private final double gearRatio;
+
+	MechanismSimulation(double gearRatio) {
+		this.gearRatio = gearRatio;
+	}
```

Comment:
This could be solved with a function declaration to get the gearRatio

Reply:
So then you copy this constructor to each class that extends this one with the exact same code?